### ISSUE TITLE
Not saving all the checkpoints at the end of every epoch

### ISSUE THREAD

**Issue by samvam24** [PyTorchLightning/pytorch-lightning#2157]
I am unable to save checkpoints at the end of validation epoch.
My trainer code:
```python   
trainer = pl.Trainer(
        max_epochs=max_epochs,
        gpus=gpus,
        distributed_backend=args.distributed_backend,
        save_top_k=10,
        check_val_every_n_epoch=1,
        show_progress_bar=True,
    )
```

Standard validation code at end of epoch:

```python   
def validation_epoch_end(self, outputs):
        mean_loss = torch.stack([x["val_loss"] for x in outputs]).mean()
        tensorboard_logs = {"val_mean_loss": mean_loss}
        return {"val_loss": mean_loss, 'log': tensorboard_logs}
```
The lightning_logs directory is like:
```
lightning_logs/
├── version_0
│   ├── checkpoints
│   ├── events.out.tfevents.1591963200.sam.2187.0
│   └── hparams.yaml
├── version_1
│   └── checkpoints
├── version_2
│   ├── checkpoints
│   │   └── epoch=1.ckpt
│   ├── events.out.tfevents.1591963715.sam.2559.0
│   └── hparams.yaml
└── version_3
    └── checkpoints
```
There is only one checkpoint file being saved.
Pytorch-lightning: 0.7.6, pytorch: 1.4.0, cuda: 10.1

**Comment by pvnieo** [comment]
what is your `validation_step` function?

**Comment by samvam24** [comment]
criteria is binary cross entropy loss.
```python
    def validation_step(self, batch, batch_idx):
        X, y = batch
        output = self(*X)
        loss = self.criteria(output, y)
        tensorboard_logs = {"val_loss": loss}
        return {"val_loss": loss, 'log': tensorboard_logs}
```

**Comment by pvnieo** [comment]
Try use a checkpoint callback:
```python
checkpoint_callback = pl.callbacks.ModelCheckpoint(
        monitor="val_loss",
        mode="min",
        save_last=True,
        save_top_k=10,
        verbose=False,
    )
```
and add to trainer:
```python
trainer = pl.Trainer(
        max_epochs=max_epochs,
        gpus=gpus,
        distributed_backend=args.distributed_backend,
        checkpoint_callback=checkpoint_callback,
    )
```

**Comment by samvam24** [comment]
Thanks @pvnieo. Just realized the default parameter`save_best = True`. Weird enough, I think my first epoch had the minimum `val_loss`. Closing the issue.
